MerlinAU.sh

```diff
@@ -2561,6 +2619,348 @@ _GetLatestFWUpdateVersionFromWebsite_()
+
+    # Extract the file extension
+    extension="${sanitized_filename##*.}"   
+
+    # Combine path, custom file name, and extension before download
+    FW_DL_FPATH="${FW_ZIP_DIR}/${FW_FileName}.${extension}"
+    FW_MD5_GITHUB="${FW_ZIP_DIR}/${FW_FileName}.md5"
+    FW_Changelog_GITHUB="${FW_ZIP_DIR}/${FW_FileName}_Changelog.txt"
+
+    # Download the firmware using the release link
+    wget --timeout=5 --tries=4 --waitretry=5 --retry-connrefused \
+         -O "$FW_DL_FPATH" "$release_link"
+    if [ ! -f "$FW_DL_FPATH" ]; then
+        return 1
+    fi
+
+    # Download the latest MD5 checksum
+    Say "Downloading latest MD5 checksum ${GRNct}${md5_url}${NOct}"
+    wget --timeout=5 --tries=4 --waitretry=5 --retry-connrefused \
+         -O "$FW_MD5_GITHUB" "$md5_url"
+    if [ ! -f "$FW_MD5_GITHUB" ]; then
+        return 1
+    fi
+
+    # Download the latest changelog
+    Say "Downloading latest Changelog ${GRNct}${Gnuton_changelogurl}${NOct}"
+    wget --timeout=5 --tries=4 --waitretry=5 --retry-connrefused \
+         -O "$FW_Changelog_GITHUB" "$Gnuton_changelogurl"
+    if [ ! -f "$FW_Changelog_GITHUB" ]; then
+        return 1
+    else
+        return 0
+    fi
+}
+
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-Apr-18] ##
+##---------------------------------------##
+_DownloadForMerlin_() {
+    
+    wget --timeout=5 --tries=4 --waitretry=5 --retry-connrefused \
+         -O "$FW_ZIP_FPATH" "$release_link"
+
+    # Check if the file was downloaded successfully
+    if [ ! -f "$FW_ZIP_FPATH" ]; then
+        return 1
+    else
+        return 0
+    fi
+}
+
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-Apr-18] ##
+##---------------------------------------##
+_UnzipMerlin_() {
+    Say "-----------------------------------------------------------"
+    # List & log the contents of the ZIP file
+    unzip -l "$FW_ZIP_FPATH" 2>&1 | \
+    while IFS= read -r uzLINE ; do Say "$uzLINE" ; done
+    Say "-----------------------------------------------------------"
+
+    # Extracting the firmware binary image
+    if unzip -o "$FW_ZIP_FPATH" -d "$FW_BIN_DIR" -x README* 2>&1 | \
+       while IFS= read -r line ; do Say "$line" ; done
+    then
+        Say "-----------------------------------------------------------"
+        #---------------------------------------------------------------#
+        # Check if ZIP file was downloaded to a USB-attached drive.
+        # Take into account special case for Entware "/opt/" paths.
+        #---------------------------------------------------------------#
+        if ! echo "$FW_ZIP_FPATH" | grep -qE "^(/tmp/mnt/|/tmp/opt/|/opt/)"
+        then
+            # It's not on a USB drive, so it's safe to delete it
+            rm -f "$FW_ZIP_FPATH"
+        elif ! _ValidateUSBMountPoint_ "$FW_ZIP_BASE_DIR" 1
+        then
+            -------------------------------------------------------------#
```

Comment:
Error in line 2808: missing number char # to indicate a comment.

Reply:
Go, go, go, go

Reply:
Yahhhhhh Martinski!!!

Reply:
Done :)